File: aqa_python/automation_playwright/pom/contactlist_page.py
from pom.base_page import BasePage

class ContactListPage(BasePage):
    def __init__(self, page):
        super().__init__(page)
        self.page_url = f"{self.config.BASE_URL}/contactList"
        self.selectors = {
            "add_contact_button": "#add-contact",
            "table": "#myTable",
            "logout_button": "#logout"
        }

    def go_to_url(self):
        self.page.goto(self.page_url)
        self.page.wait_for_timeout(200)

    def click_add_contact(self):
        self.page.click(self.selectors["add_contact_button"])
        self.page.wait_for_timeout(200)

    def click_logout(self):
        self.page.click(self.selectors["logout_button"])
        self.page.wait_for_timeout(200)

    def get_contact_list_table(self):
        self.page.wait_for_selector(self.selectors["table"])
        return self.page.query_selector(self.selectors["table"])

    def get_contact_list_table_rows(self):
        table = self.get_contact_list_table()
        self.page.wait_for_selector("tr")
        return table.query_selector_all("tr")

    def get_contact_row_by_first_name(self, first_name: str):
        rows = self.get_contact_list_table_rows()
        for row in rows:
            row_text = row.text_content()
            if first_name in row_text:
                return row
        return None

File: aqa_js/automation_playwright/pom/contactlistpage.js
const {Page, expect} = require('@playwright/test');
const BasePage = require('./basepage');

class ContactListPage extends BasePage {
    /**
     * @param {Page} page - Playwright Page object
     */
    constructor(page) {
        super(page);
        this.PageUrl = `${this.config.BASE_URL}/contactList`;
        this.ids = {
            addContactButtonId: '#add-contact',
            tableId: '#myTable',
            logoutButtonId: '#logout',
        };
    }

    async go_to_url() {
        await this.page.goto(this.PageUrl);
    }

    async click_add_contact() {
        await this.page.click(this.ids.addContactButtonId);
    }

    async click_logout() {
        await this.page.click(this.ids.logoutButtonId);
    }

    async get_contact_list_table() {
        await this.page.waitForSelector(this.ids.tableId)
        return await this.page.$(this.ids.tableId);
    }

    async get_contact_list_table_rows() {
        const table = await this.get_contact_list_table();
        await this.page.waitForSelector('tr')
        return await table.$$('tr');
    }

    async get_contact_row_by_first_name(firstName) {
        const rows = await this.get_contact_list_table_rows();
        let targetContactRow = null;

        for (const row of rows) {
            const rowText = await row.textContent();
            console.log("Row Text:", rowText);
            if (rowText.includes(firstName)) {
                targetContactRow = row;
                break;
            }
        }

        return targetContactRow;
    }
}

module.exports = ContactListPage;